fsspec/implementations/memory.py

```diff
@@ -9,19 +9,33 @@
 
 logger = logging.Logger("fsspec.memoryfs")
 
+_global_store = {}
+_global_pseudo_dirs = [""]
+
```

Comment:
If backward compatibility wouldn't be a concern, I would rather just make `MemoryFileSystem` not use global store at all and would create a global memfs instance that people could use. But oh well 🤷